src/openms/source/ANALYSIS/OPENSWATH/PeakIntegrator.cpp

```diff
@@ -354,4 +321,38 @@ namespace OpenMS
       std::max(peak_apex_pos - psm.start_position_at_10, psm.end_position_at_10 - peak_apex_pos);
     return psm;
   }
+
+  template <typename PeakContainerConstIteratorT>
+  double PeakIntegrator::findPosAtPerc_(
+    PeakContainerConstIteratorT it_begin,
+    PeakContainerConstIteratorT it_end,
+    const double peak_height,
+    const double perc,
+    const bool is_left_half
+  ) const
+  {
+    const double goal_intensity = peak_height * perc;
+    PeakContainerConstIteratorT closest = is_left_half ? it_begin : it_end - 1; // TODO assuming enough points here
+    if (is_left_half)
+    {
+      for (PeakContainerConstIteratorT it = it_begin; it != it_end; ++it)
+      {
+        if (it->getIntensity() <= goal_intensity)
+        {
+          closest = it;
+        }
+      }
+    }
+    else
+    {
+      for (PeakContainerConstIteratorT it = it_end - 1; it != it_begin; --it)
```

Comment:
This is an important change that we are iterating from right to left for the right side of the peak.  Nice.